### PR TITLE
Chat 111 segment proxy

### PR DESCRIPTION
**Pull request by raimondi1337**

### CODE REVIEW

**Comment by raimondi1337** [comment]
Functions Code
https://gist.github.com/raimondi1337/7940795e4fa0e5db7789f22e22be1e35

**Comment by MaxPresman** [comment]
@raimondi1337 there are additional calls inside the javascript files

File: setup/src/script.js [hunk 1]
```diff
@@ -77,7 +104,34 @@ module.exports = class {
...
+                        url: location.href,
+                        title: document.title,
+                        search: location.search,
+                        referrer: document.referrer
+                    },
+                    userAgent: navigator.userAgent
+                },
+                userId: this.userId
+            };
+            $.ajax({
```
Comment (MaxPresman) [inline setup/src/script.js]: this needs to be extracted into a function to avoid duplication.

File: setup/src/script.js [hunk 2]
```diff
@@ -41,7 +41,34 @@ module.exports = class {
         if (this.userId && tokenCookie) {
             this.provisionElement.show();
             this.loginElement.hide();
-            analytics.identify(this.userId);
+
+            // analytics.identify(this.userId);
+            let analyticsData = {
```
Comment (MaxPresman) [inline setup/src/script.js]: analytics data looks duplicate, can we extract it into a common method?

File: setup/index.html [hunk 3]
```diff
@@ -20,10 +20,39 @@
...
-      analytics.load("5ANv64jDz5MgcidXwijzUDF8g9PhKRqz"); // prod => OdINAt6BhYOBsN2n1ulSDi5S9SipTaqu
-      analytics.page();
-      }}();
+    // analytics.page();
+    let analyticsData = {
+        type: 'page',
+        anonymousId: document.cookie.substring(document.cookie.indexOf('=')+4,document.cookie.indexOf(';')-3),
```
Comment (ajb413) [inline setup/index.html]: is this specific substring required? maybe a random uuid would be better?

**Pull request merged**